cpp/core/utils/compression.h

```diff
@@ -33,11 +33,43 @@
 
 namespace gluten {
 
-#if defined(GLUTEN_ENABLE_QAT) or defined(GLUTEN_ENABLE_IAA)
+#if defined(GLUTEN_ENABLE_QAT)
+static const std::vector<arrow::Compression::type> kSupportedCodec = {
+    arrow::Compression::LZ4_FRAME,
+    arrow::Compression::ZSTD,
+    arrow::Compression::GZIP};
+inline Result<std::unique_ptr<arrow::util::Codec>> createCodec(arrow::Compression::type compressedType) {
+  std::unique_ptr<arrow::util::Codec> codec;
+  switch (compressedType) {
+    case arrow::Compression::LZ4_FRAME:
+      return arrow::util::Codec::Create(compressedType);
+    case arrow::Compression::ZSTD:
+      codec = qat::makeDefaultQatZstdCodec();
```

Comment:
This will ignore "codecBackend" when creating native writer. Here we want to create software codec but not qat codec if the "codecBackend" is null.